olp-cpp-sdk-dataservice-read/src/repositories/PrefetchTilesRepository.cpp

```diff
@@ -314,13 +311,60 @@ SubQuadsResult PrefetchTilesRepository::FilterSkippedTiles(
       ++sub_quad_it;
     }
   }
-  if (request_only_input_tiles) {
+
+  return sub_tiles;
+}
+
+SubQuadsResult PrefetchTilesRepository::FilterTilesByList(
+    const PrefetchTilesRequest& request, SubQuadsResult sub_tiles) {
+  const bool aggregation_enabled = request.GetAggregationEnabled();
+
+  const auto& tile_keys = request.GetTileKeys();
+
+  if (!aggregation_enabled) {
+    for (auto it = sub_tiles.begin(); it != sub_tiles.end();) {
+      if (std::find(tile_keys.begin(), tile_keys.end(), it->first) ==
+          tile_keys.end()) {
+        it = sub_tiles.erase(it);
+      } else {
+        ++it;
+      }
+    }
+
     for (const auto& tile : tile_keys) {
       if (sub_tiles.find(tile) == sub_tiles.end()) {
         sub_tiles[tile] = "";
       }
     }
+
+  } else {
+    SubQuadsResult result;
+
+    auto append_tile = [&](const geo::TileKey& key) {
+      auto sub_tile_it = sub_tiles.find(key);
+      if (sub_tile_it != sub_tiles.end()) {
+        result[sub_tile_it->first] = sub_tile_it->second;
+        return true;
+      } else {
+        return false;
+      }
+    };
+
+    for (const auto& tile : tile_keys) {
+      auto aggregated_tile = tile;
+
+      while (aggregated_tile.IsValid() && !append_tile(aggregated_tile)) {
+        aggregated_tile = aggregated_tile.Parent();
+      }
```

Comment:
During `query` we don't do any aggregated search, so why it's expected `sub_tiles` has closest parent?

Reply:
Tree index contains both children and parent metadata.